cosmos/client.go

```diff
@@ -129,3 +131,34 @@ func (c *Client) BuildAndBroadcastMsg(msg sdktypes.Msg, accName, accPassword str
 		return nil, errors.New("i/o timeout")
 	}
 }
+
+// Stream subscribes to the provided query and returns the hash of the matching ressources.
+func (c *Client) Stream(query string) (chan hash.Hash, error) {
+	subscriber := xstrings.RandAsciiLetters(8)
+	eventStream, err := c.Subscribe(context.Background(), subscriber, query)
+	if err != nil {
+		return nil, err
+	}
+
+	hashC := make(chan hash.Hash)
+	go func() {
+		for event := range eventStream {
+			ressHashes := event.Events[EventHashType]
+			if len(ressHashes) != 1 {
+				// or panic(err) - grpc api do not support
```

Comment:
could we return those error on the chan?
either with a new chan or a struct that contain both the hash and the error.

Reply:
Yes, but then what to do with them? Retrun back to client?

Reply:
> Retrun back to client?

yes